src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -183,6 +183,25 @@ static Value builtinStringSubstr(ExecutionState& state, Value thisValue, size_t
 
 static Value builtinStringMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
 {
+#ifdef ESCARGOT_ENABLE_ES2015
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, Object, match);
+    Value argument = argv[0];
+    RegExpObject* regexp;
+
+    if (argument.isUndefinedOrNull() == false) {
+        Value match = obj->getMethod(state, argument, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().match));
+        if (match.isUndefined() == false) {
+            return FunctionObject::call(state, match, argument, 1, &thisValue);
+        }
+    }
+
+    String* str = Value(obj).toString(state);
+    regexp = new RegExpObject(state, argument.isUndefined() ? String::emptyString : argument.toString(state), String::emptyString);
+    Value func = regexp->getMethod(state, regexp, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().match));
```

Comment:
You must invoke the `@@match` routine, so the `getMethod` call is incorrect here.

Reply:
This problem is still valid.